mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -2026,14 +2052,14 @@ def search_logged_models(
         self,
         experiment_ids: list[str],
         filter_string: Optional[str] = None,
-        datasets: Optional[list[DatasetInput]] = None,
+        datasets: Optional[list[dict[str, Any]]] = None,
         max_results: Optional[int] = None,
         order_by: Optional[list[dict[str, Any]]] = None,
         page_token: Optional[str] = None,
     ) -> PagedList[LoggedModel]:
-        if datasets:
+        if datasets and not all("dataset_name" in dataset for dataset in datasets):
```

Comment:
```suggestion
        if datasets and not all(d.get("dataset_name") for d in datasets):
```

To reject None and an empty string.

Reply:
Sounds good, added to file store as well